### PR TITLE
refactor: use monkeypatch to replace mock library patch blocks

### LINKED ISSUE #673: Replace legacy 'mock' functionality with pytest's monkeypatch.
## Issue description / feature objectives
Currently the trestle codebase for patching is primarily using functionality from the 'mock' library. Pytest provides functionality which is equivalent but cleaner using `monkeypatch` see: https://docs.pytest.org/en/latest/how-to/monkeypatch.html

This provides some substantial advantages - principally it avoids the 'with' argument that can be prone to being used incorrectly (scope can affect asserts used to test conditions).

## Completion criteria:
All use of mock patch are removed in favour of pytest intrinsic functionality.

### PR DESCRIPTION
**Pull request by jeffdmgit**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] All commits are signed-off.

This improves all patch blocks (from the *mock* library) with *monkeypatch* to avoid the indentations and subsequent asserts -- except for one instance in `test_utils.py`, which would require adding a *MonkeyPatch* arg to all calls to `test_utils.create_trestle_project_with_model()`.

This closes #673.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -148,9 +149,9 @@ def tmp_trestle_dir(tmp_path: pathlib.Path) -> Iterator[pathlib.Path]:
     pytest_cwd = pathlib.Path.cwd()
     os.chdir(tmp_path)
     testargs = ['trestle', 'init']
-    with patch.object(sys, 'argv', testargs):
-        # FIXME: Correctly capture return codes
-        Trestle().run()
+    monkeypatch.setattr(sys, 'argv', testargs)
+    # FIXME: Correctly capture return codes
+    Trestle().run()
```
Comment (butler54) [inline tests/conftest.py:154]: Worthwhile fixing this now we are editing the code.

File: tests/trestle/core/commands/import__test.py [hunk 2]
```diff
@@ -244,34 +257,41 @@ def test_import_bad_input_extension(tmp_trestle_dir: pathlib.Path) -> None:
...
     """Test model load failures."""
+
+    def load_file_mock(*args, **kwargs):
+        raise err.TrestleError('stuff')
+
+    def load_file_permission_error_mock(*args, **kwargs):
+        raise PermissionError
+
+    def load_file_json_decode_error_mock(*args, **kwargs):
+        raise JSONDecodeError(msg='Extra data:', doc=bad_file.name, pos=0)
```
Comment (butler54) [inline tests/trestle/core/commands/import__test.py:270]: You can probably simplify this with lambda functions if need be as well.

File: tests/trestle/core/commands/remove_test.py [hunk 3]
```diff
@@ -233,13 +242,25 @@ def test_run_failure_filenotfounderror(tmp_path, sample_catalog_minimal):
...
+    exitcode = Trestle().run()
+    assert exitcode == 1
 
 
-def test_run_failure_plan_execute(tmp_path, sample_catalog_minimal):
+def test_run_failure_plan_execute(
+    tmp_path: pathlib.Path, sample_catalog_minimal: Catalog, monkeypatch: MonkeyPatch, caplog: pytest.LogCaptureFixture
+):
     """Test failure plan execute() in _run on RemoveCmd."""
+    # Plant this specific logged error for failing execution in mock_execute:
```
Comment (butler54) [inline tests/trestle/core/commands/remove_test.py:254]: This caplog test is neat. Does it capture both stdout and stderr?

**Review by butler54 — CHANGES_REQUESTED** [review]
LGTM excluding the fixme that has been skipped in conftest.py

File: tests/trestle/core/commands/remove_test.py (hunk 3 above)
Comment (jeffdmgit) [reply tests/trestle/core/commands/remove_test.py:254]: There seems to be *capsys* for both stdout and stderr.

Comment (butler54) [reply tests/trestle/core/commands/remove_test.py:254]: Nice!

File: tests/conftest.py (hunk 1 above)
Comment (butler54) [reply tests/conftest.py:154]: As discussed  leaving for current PR pending #745

**Review by butler54 — APPROVED** [review]

**Pull request merged**